src/Vault.sol

```diff
@@ -247,60 +356,84 @@ contract Vault is IVault, ERC20PermitUpgradeable, AccessControlUpgradeable, Reen
             _spendAllowance(owner, caller, shares);
         }
 
-        IStrategy(vaultStorage.bufferStrategy).withdraw(assets, address(this), address(this));
+        IStrategy(vaultStorage.buffer).withdraw(assets, address(this), address(this));
 
         SafeERC20.safeTransfer(IERC20(asset_), receiver, assets);
 
         _burn(owner, shares);
         emit Withdraw(caller, receiver, owner, assets, shares);
     }
 
-    function _convertToAssets(address asset_, uint256 shares, Math.Rounding rounding) internal view returns (uint256) {
-        uint256 baseDenominatedShares = _convertAssetToBase(asset_, shares);
-        return baseDenominatedShares.mulDiv(totalSupply() + 10 ** _decimalsOffset(), totalAssets() + 1, rounding);
-    }
-
-    function _convertToShares(address asset_, uint256 assets_, Math.Rounding rounding)
+    /**
+     * @notice Internal function to convert vault shares to the base asset.
+     * @param asset_ The address of the asset.
+     * @param shares The amount of shares to convert.
+     * @param rounding The rounding direction.
+     * @return (uint256 assets, uint256 baseAssets) The equivalent amount of assets.
+     */
+    function _convertToAssets(address asset_, uint256 shares, Math.Rounding rounding)
         internal
         view
-        returns (uint256)
+        returns (uint256, uint256)
     {
-        uint256 convertedAssets = _convertAssetToBase(asset_, assets_);
-        return convertedAssets.mulDiv(totalSupply() + 10 ** _decimalsOffset(), totalAssets() + 1, rounding);
-    }
-
-    function _convertAssetToBase(address asset_, uint256 amount) internal view returns (uint256) {
-        uint256 rate = IRateProvider(rateProvider()).getRate(asset_);
-        return amount.mulDiv(rate, 10 ** getAsset(asset_).decimals, Math.Rounding.Floor);
-    }
-
-    // function _convertBaseToAsset(address asset_, uint256 baseAmount) internal view returns (uint256) {
-    //     uint256 rate = IRateProvider(rateProvider()).getRate(asset_);
-    //     return baseAmount.mulDiv(10 ** getAsset(asset_).decimals, rate, Math.Rounding.Floor);
-    // }
-
-    function _decimalsOffset() internal pure returns (uint8) {
-        return 0;
-    }
-
+        uint256 assets = shares.mulDiv(totalAssets() + 1, totalSupply() + 10 ** 0, rounding);
+        uint256 baseAssets = _convertAssetToBase(asset_, assets);
```

Comment:
this still needs to be fixed with a function that does _covertBaseToAsset(asset_, assets)